### ISSUE TITLE
Deprecated github v2 api = no access to Jenkins

### ISSUE THREAD

**Issue by ggoodale** [mocleiri/github-oauth-plugin#24]
Today @github turned off the v2 API, which means any jenkins instance using this plugin for auth can't authenticate users:

```
java.io.FileNotFoundException: https://github.com/api/v2/json/user/show?access_token=xxxxxxxx
79
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1401)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
        at org.kohsuke.github.GitHub._retrieve(GitHub.java:204)
        at org.kohsuke.github.GitHub.retrieveWithAuth(GitHub.java:181)
        at org.kohsuke.github.GitHub.retrieveWithAuth(GitHub.java:173)
        at org.kohsuke.github.GitHub.getMyself(GitHub.java:248)
        at org.kohsuke.github.GitHub.<init>(GitHub.java:108)
        at org.kohsuke.github.GitHub.connectUsingOAuth(GitHub.java:139)
        at org.jenkinsci.plugins.GithubAuthenticationToken.<init>(GithubAuthenticationToken.java:68)
        at org.jenkinsci.plugins.GithubSecurityRealm.doFinishLogin(GithubSecurityRealm.java:312)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:616)
        at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:288)
        at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:151)
        at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:90)
        at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:111)
        at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:574)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:659)
        at org.kohsuke.stapler.MetaClass$4.doDispatch(MetaClass.java:203)
        at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:53)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:574)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:659)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:488)
        at org.kohsuke.stapler.Stapler.service(Stapler.java:162)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:45)
       ...
```

**Comment by mattrobenolt** [comment]
:+1:

**Comment by naaman** [comment]
+1

**Comment by ryanbrainard** [comment]
+1

**Comment by mattrobenolt** [comment]
If anyone has any Java chops, the underlying GitHub API has v3 support already: https://github.com/kohsuke/github-api I'm just a measly Python guy. :(

**Comment by greylurk** [comment]
+1 here.

**Comment by kbond** [comment]
yikes, how can I disable the plugin so I can login?

**Comment by jhedstrom** [comment]
:+1:

**Comment by mattrobenolt** [comment]
You can just delete it out of your plugins directory, then restart Jenkins. 

## 

Matt Robenolt
@mattrobenolt

**Comment by jhedstrom** [comment]
Also not a Java dev, but I wonder if it's as simple as bumping the version and (somehow) rebuilding dependencies? https://github.com/mocleiri/github-oauth-plugin/blob/master/pom.xml#L67

**Comment by scribu** [comment]
> You can just delete it out of your plugins directory, then restart Jenkins.

You might also need to manually edit config.xml, commenting out the `authorizationStrategy` tag.

**Comment by mattrobenolt** [comment]
I didn't have to do that, it worked fine when I just deleted it. I don't know about for others though.

## 

Matt Robenolt
@mattrobenolt

**Comment by kbond** [comment]
just deleting and restarting worked for me as well

**Comment by ryanbrainard** [comment]
@jhedstrom, I just tried bumping the version, but unfortunately it's not that easy. The underlying libraries is a bit of a mess with mixing v2 and v3 for getting org info. I'm going to see if I can get something working though so I can get our Jenkins back up and will share if I have something working.

**Comment by jayzes** [comment]
I was just going to say, looks like the organization retrieval call is still using the V2 API:

https://github.com/kohsuke/github-api/blob/master/src/main/java/org/kohsuke/github/GitHub.java#L185

**Comment by ryanbrainard** [comment]
Ok, I fixed the calls for getting org info and it is working again. You can download the artifact here:
https://github.com/ryanbrainard/github-oauth-plugin/downloads

Installation is like this:

```
cd $JENKINS_HOME/plugins
rm -rf github-oauth github-oauth.hpi
curl -O --location https://github.com/downloads/ryanbrainard/github-oauth-plugin/github-oauth-0.12-RB-SNAPSHOT.hpi
```

Then restart Jenkins.

I changed as little as possible to get it working, but there are still issues that need fixing to clean up the v2 vs v3 use; however, it looks like @kohsuke was working on [that](https://github.com/kohsuke/github-api/commit/b5f7208b0dd65b353483628a4ee259b701746b2b) as I was working on my fix. We can probably reconcile these two soon, but this should get you back up and running in the time being.

**Comment by scribu** [comment]
It works! Thanks, @ryanbrainard.

**Comment by scribu** [comment]
Note that you have to `cd $JENKINS_HOME/plugins` not `cd $JENKINS_HOME`.

**Comment by ryanbrainard** [comment]
@scribu, thanks for the correction. I just updated the instructions.

**Comment by sborsje** [comment]
@ryanbrainard Thanks, works like a charm!

**Comment by ryanbrainard** [comment]
I just sent a [pull request](https://github.com/kohsuke/github-api/pull/10) to @kohsuke to fix some things the `github-oauth-plugin` uses in his `github-api` library. Once that's in, it really will be just a dependency version bump for the  `github-oauth-plugin`, so hopefully we can have this as a real plugin upgrade. For now, glad to hear my band-aid release is working for people :)

**Comment by jhedstrom** [comment]
Thanks @ryanbrainard!

**Comment by harishreddyreddy** [comment]
Thanks @ryanbrainard

**Comment by tlfbrito** [comment]
@ryanbrainard Thanks, man! You just save my day!

**Comment by minimal** [comment]
Thanks @ryanbrainard :+1:

**Comment by kohsuke** [comment]
I've released github-oauth plugin v0.12 to fix this.

It should be available in your update center shortly, but in the mean time, you can download it from here: http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/github-oauth/0.12/github-oauth-0.12.hpi

**Comment by kohsuke** [comment]
Actually, what you need is http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/github-api/1.28/github-api-1.28.hpi

**Comment by nextmat** [comment]
Thanks guys!

**Comment by TALlama** [comment]
For ease of copy/paste, putting @ryanbrainard and @kohsuke's stuff together:

```
cd $JENKINS_HOME/plugins
rm -rf github-oauth github-oauth.hpi
curl -O --location http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/github-api/1.28/github-api-1.28.hpi
```

Then restart Jenkins however you do (mine is a `sudo service jenkins restart`)

**Comment by harmaty** [comment]
After doing this, I've got following error:

hudson.security.AccessDeniedException2: anonymous is missing the Read permission
    at hudson.security.ACL.checkPermission(ACL.java:53)
    at hudson.model.Node.checkPermission(Node.java:381)
    at jenkins.model.Jenkins.getTarget(Jenkins.java:3363)
    at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:496)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:646)
    at org.kohsuke.stapler.Stapler.invoke(Stapler.java:477)
    at org.kohsuke.stapler.Stapler.service(Stapler.java:159)

**Comment by flyaruu** [comment]
Thanks guys, you've saved my day

**Comment by harmaty** [comment]
Downloaded http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/github-oauth/0.12/github-oauth-0.12.hpi  instead of 
http://repo.jenkins-ci.org/public/org/jenkins-ci/plugins/github-api/1.28/github-api-1.28.hpi
and now it works fine

**Comment by kbond** [comment]
I deleted the github-oauth plugin manually the other day.  Now it is not showing up in the available plugins list... how to I get it back? or do I have to install manually?

**Comment by bzenkus-radialnetwork** [comment]
Thanks all.
 I updated both (github-api-1.28 and github-oauth-0.12.hpi) - working great now.
The api fixed a UI issue I had where the Jenkins top banner, icons and "Log in" did not display if not logged in.
The oauth-0.12 corrected issue #24 for me.
Would be nice to have known this sooner... occurred right after we moved Jenkins to a different cloud environment.
Any way to subscribe to Jenkins/Git issues?
